src/invidious/routes/before_all.cr

```diff
@@ -100,6 +88,24 @@ module Invidious::Routes::BeforeAll
       end
     end
 
+    unregistered_path_whitelist = {"/", "/login", "/licenses", "/privacy"}
+    if CONFIG.login_only && !env.get?("user") && !unregistered_path_whitelist.includes?(env.request.path)
+      env.response.headers["Location"] = "/login"
+      haltf env, status_code: 302
```

Comment:
I'm also not entirely sure about this part here. Since access is denied, because of a missing authentication, a `401`should be returned.

But sending a redirect somewhat defeats the purpose of returning a `401`.

Maybe a second configuration option should be added, like: `redirect_login_only`.

If `true` a `302`is returned and the user is redirected to `/login` otherwise, a `401`is sent and the nothing happens.